recipes/ignition-tools/all/conanfile.py

```diff
@@ -88,18 +99,59 @@ def package(self):
 
+        self._create_cmake_module_variables(
+            os.path.join(self.package_folder, self._module_file_rel_path),
+                tools.Version(self.version))
+
+    @staticmethod
+    def _create_cmake_module_variables(module_file, version):
+        content = textwrap.dedent("""\
+            set(IGN_TOOLS_CONAN_PACKAGE_DIR ${{CMAKE_CURRENT_LIST_DIR/../..}})
+            list(APPEND IGNITION-TOOLS_BINARY_DIRS ${{IGN_TOOLS_CONAN_PACKAGE_DIR}}/bin)
+            list(APPEND IGNITION-TOOLS_INCLUDE_DIRS ${{IGN_TOOLS_CONAN_PACKAGE_DIR}}/include)
+            list(APPEND IGNITION-TOOLS_LIBRARY_DIRS ${{IGN_TOOLS_CONAN_PACKAGE_DIR}}/libs)
+            list(APPEND IGNITION-TOOLS_CFLAGS -I/${{IGN_TOOLS_CONAN_PACKAGE_DIR}}/include)
+            list(APPEND IGNITION-TOOLS_CXX_FLAGS -std=c++11)
+            set(ignition-tools{major}_VERSION_MAJOR {major})
+            set(ignition-tools{major}_VERSION_MINOR {minor})
+            set(ignition-tools{major}_VERSION_PATCH {patch})
+            set(ignition-tools{major}_VERSION_STRING "{major}.{minor}.{patch}")
+        """.format(major=version.major, minor=version.minor, patch=version.patch))
+        tools.save(module_file, content)
+
     def package_info(self):
-        version_major = tools.Version(self.version).major
-        self.cpp_info.names["cmake_find_package"] = "ignition-tools{}".format(version_major)
-        self.cpp_info.names["cmake_find_package_multi"] = "ignition-tools{}".format(version_major)
-
-        self.cpp_info.components["libignition-tools"].libs = ["ignition-tools-backward"]
-        self.cpp_info.components["libignition-tools"].includedirs.append("include/ignition/tools{}".format(version_major))
-        self.cpp_info.components["libignition-tools"].names["cmake_find_package"] = "ignition-tools{}".format(version_major)
-        self.cpp_info.components["libignition-tools"].names["cmake_find_package_multi"] = "ignition-tools{}".format(version_major)
-        self.cpp_info.components["libignition-tools"].names["pkg_config"] = "ignition-tools{}".format(version_major)
+        lib_name = "ignition-tools"
+        self.cpp_info.names["cmake_find_package"] = lib_name
+        self.cpp_info.names["cmake_find_package_multi"] = lib_name
+        self.cpp_info.names["cmake_paths"] = lib_name
+        self.cpp_info.set_property("cmake_file_name", "ignition-tools")
+        self.cpp_info.components["core"].names["cmake_find_package"] = "core"
+        self.cpp_info.components["core"].names["cmake_find_package_multi"] = "core"
+        self.cpp_info.components["core"].names["cmake_paths"] = "core"
+        self.cpp_info.components["core"].bindirs = ["bin"]
+
+        self.cpp_info.components["core"].libs = []
+        self.cpp_info.components["core"].includedirs = []
+        if int(tools.Version(self.version).minor) > 2:
+            self.cpp_info.components["core"].libs.append(lib_name +"-backward")
```

Comment:
I found this https://github.com/gazebosim/gz-tools/blob/9881cd0933acc5546cb0b4d686b13cd210cbb5f0/src/CMakeLists.txt#L28

But they refer to is as "backwards" not "core" so I am little confused why you called it core?

Reply:
Lastly I notice that this builds against a vendored library `backwards-hpp` https://github.com/gazebosim/gz-tools/commit/fd3e828e28e8f0f3634a4c9772a630576dfa689c#diff-7208973e7d059ad1b4f2a900850ed5bb72b0cd510c85b1d33d904f67c70123d9R1

and it's even available from Conan... ideal this should come from Conan so there isnt any surprise in the graph

Reply:
@prince-chrismc  I did not get the last comment. I don't find a backward.hpp file in the `package` directory of ignition-tools. Or am I missing something ?